go/payload-commit/service/service.go

```diff
@@ -0,0 +1,578 @@
+	if err != nil {
+		log.WithError(err).Error("failed to add snapshotter status report to redis")
+
+		return err
+	}
+
+	return nil
+}
+
+func (s *PayloadCommitService) uploadToIPFSandW3s(msg *datamodel.PayloadCommitMessage) error {
+	log.WithField("msg", msg).Debug("uploading payload commit msg to ipfs and web3 storage")
+
+	wg := sync.WaitGroup{}
+
+	wg.Add(1)
+
+	// upload to ipfs
+	var ipfsUploadErr error
+
+	go func() {
+		defer wg.Done()
+
+		ipfsUploadErr = backoff.Retry(func() error {
+			ipfsUploadErr = s.ipfsClient.UploadSnapshotToIPFS(msg)
+			if ipfsUploadErr != nil {
+				log.WithError(ipfsUploadErr).Error("failed to upload snapshot to ipfs, retrying")
+
+				return ipfsUploadErr
+			}
+
+			return nil
+		}, backoff.NewExponentialBackOff())
+
+		if ipfsUploadErr != nil {
+			log.WithError(ipfsUploadErr).Error("failed to upload snapshot to ipfs after max retries")
+		}
+	}()
+
+	// upload to web3 storage
+	var (
+		w3sUploadErr error
+		snapshotCid  string
+	)
+
+	if msg.Web3Storage {
+		wg.Add(1)
+
+		go func() {
+			wg.Done()
+
+			w3sUploadErr = backoff.Retry(func() error {
+				snapshotCid, w3sUploadErr = s.web3sClient.UploadToW3s(msg.Message)
+				if w3sUploadErr != nil {
+					return w3sUploadErr
+				}
+
+				return nil
+			}, backoff.NewExponentialBackOff())
+
+			if w3sUploadErr != nil {
+				log.WithError(w3sUploadErr).Error("failed to upload snapshot to web3 storage after max retries")
+			}
+		}()
+	}
+
+	wg.Wait()
+
+	if ipfsUploadErr != nil && !msg.Web3Storage {
+		return fmt.Errorf("failed to upload to ipfs")
+	}
+
+	if ipfsUploadErr != nil && w3sUploadErr != nil {
+		return fmt.Errorf("failed to upload to ipfs and web3 storage")
+	}
+
+	if ipfsUploadErr != nil && w3sUploadErr == nil {
+		msg.SnapshotCID = snapshotCid
+
+		return nil
+	}
+
+	return nil
+}
+
+// signPayload signs the payload commit message for relayer.
+func (s *PayloadCommitService) signPayload(snapshotCid, projectId string, epochId int64) (*apitypes.TypedData, []byte, error) {
+	log.Debug("signing payload commit message")
+
+	signerData, err := signer.GetSignerData(s.ethClient, snapshotCid, projectId, epochId)
+	if err != nil {
+		log.WithError(err).Error("failed to get signer data")
+
+		go s.slackNotifier.NotifySlackWorkflow(map[string]interface{}{
+			"msg":         "failed to get signer data",
+			"error":       err.Error(),
+			"snapshotCid": snapshotCid,
+			"projectId":   projectId,
+			"epochId":     epochId,
+		}, slackutils.SeverityCritical, ServiceName)
+
+		return nil, nil, err
+	}
+
+	signature, err := signer.SignMessage(s.privKey, signerData)
+	if err != nil {
+		log.WithError(err).Error("failed to sign message")
+
+		go s.slackNotifier.NotifySlackWorkflow(map[string]interface{}{
+			"msg":        "failed to sign message",
+			"error":      err.Error(),
+			"signerData": signerData,
+		}, slackutils.SeverityCritical, ServiceName)
+
+		return nil, nil, err
+	}
+
+	verified := signer.VerifySignature(signature, signerData)
```

Comment:
What is the benefit of verifying using your own local code? It doesn't confirm much, if you really want to verify the signature then you should be using the `verify` function from contract.

P.S. - This part can be optional

Reply:
Removed local verification